src/backend/distributed/shared_library_init.c

```diff
@@ -1106,6 +1115,23 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomStringVariable(
+		"citus.hide_shards_from_app_name_prefixes",
+		gettext_noop("If application_name starts with one of these values, hide shards"),
+		gettext_noop("Citus places distributed tables and shards in the same schema. "
+					 "That can cause confusion when inspecting the list of tables on "
+					 "a node with shards. This GUC can be used to hide the shards from "
+					 "pg_class for certain applications based on the application_name "
+					 "of the connection. This behaviour can be overridden using the "
+					 "citus.override_table_visibility setting"),
+		&HideShardsFromAppNamePrefixes,
+		"psql,pgAdmin,pg_dump",
```

Comment:
Will change default to *. I played with show_shards_to_app_name_prefixes, but it seemed unituitive to do something special by default and then provide a GUC to selectively do something normal, rather than providing a GUC for doing something special.